### DISCUSSION TITLE
Strange packing algorithm behavior

### DISCUSSION THREAD

**Discussion by frankvw2017** [dvdoug/BoxPacker#538] [Q&A] [answered]
I'm putting BoxPacker 3.11.0 through its paces and I'm impressed! Thank you for making it!!

However, the packing algorithm does something I can't understand: it packs some items into a second box while the first one has ample space left for the items that go into the second box. The first box is a Stock 5 Single Wall box, 30 x 45 x 30cm (LxWxH) inside; 31 x 46 x 32cm (LxWxH) outside, maximum weight 15kg. The combined weight of this box and its contents it is 2.46kg. The box is 85.1% full.

Plugging the following into the visualizer illustrates what's happening:

`[{"box":{"reference":"Stock 5 Single Wall","innerWidth":30,"innerLength":45,"innerDepth":30},"items":[{"x":0,"y":0,"z":0,"width":30,"length":25,"depth":25,"item":{"description":"Kajigger","width":30,"length":25,"depth":25,"keepFlat":false}},{"x":0,"y":25,"z":0,"width":8,"length":10,"depth":20,"item":{"description":"Doohickey","width":8,"length":10,"depth":20,"keepFlat":true}},{"x":0,"y":0,"z":25,"width":15,"length":20,"depth":5,"item":{"description":"Gadget","width":15,"length":20,"depth":5,"keepFlat":true}},{"x":15,"y":0,"z":25,"width":15,"length":20,"depth":5,"item":{"description":"Gadget","width":15,"length":20,"depth":5,"keepFlat":true}},{"x":8,"y":25,"z":25,"width":15,"length":20,"depth":5,"item":{"description":"Gadget","width":15,"length":20,"depth":5,"keepFlat":true}},{"x":8,"y":25,"z":20,"width":15,"length":20,"depth":5,"item":{"description":"Gadget","width":15,"length":20,"depth":5,"keepFlat":true}},{"x":8,"y":25,"z":15,"width":15,"length":20,"depth":5,"item":{"description":"Gadget","width":15,"length":20,"depth":5,"keepFlat":true}},{"x":8,"y":25,"z":10,"width":15,"length":20,"depth":5,"item":{"description":"Gadget","width":15,"length":20,"depth":5,"keepFlat":true}},{"x":8,"y":25,"z":5,"width":15,"length":20,"depth":5,"item":{"description":"Gadget","width":15,"length":20,"depth":5,"keepFlat":true}},{"x":8,"y":25,"z":0,"width":15,"length":20,"depth":5,"item":{"description":"Gadget","width":15,"length":20,"depth":5,"keepFlat":true}},{"x":23,"y":25,"z":0,"width":5,"length":5,"depth":30,"item":{"description":"Whatsanamie","width":5,"length":5,"depth":30,"keepFlat":false}},{"x":0,"y":20,"z":25,"width":30,"length":5,"depth":5,"item":{"description":"Whatsanamie","width":5,"length":5,"depth":30,"keepFlat":false}},{"x":0,"y":25,"z":20,"width":8,"length":5,"depth":1,"item":{"description":"Whatzit","width":5,"length":8,"depth":1,"keepFlat":false}},{"x":0,"y":25,"z":21,"width":8,"length":5,"depth":1,"item":{"description":"Whatzit","width":5,"length":8,"depth":1,"keepFlat":false}},{"x":0,"y":25,"z":22,"width":8,"length":5,"depth":1,"item":{"description":"Whatzit","width":5,"length":8,"depth":1,"keepFlat":false}},{"x":0,"y":25,"z":23,"width":8,"length":5,"depth":1,"item":{"description":"Whatzit","width":5,"length":8,"depth":1,"keepFlat":false}},{"x":1,"y":25,"z":24,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":5,"y":30,"z":23,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":2,"y":30,"z":26,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":3,"y":30,"z":20,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":3,"y":30,"z":23,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":3,"y":30,"z":26,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":4,"y":30,"z":20,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":4,"y":30,"z":23,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":4,"y":30,"z":26,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":5,"y":30,"z":20,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":6,"y":30,"z":26,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":5,"y":30,"z":26,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":6,"y":30,"z":20,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":6,"y":30,"z":23,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":2,"y":30,"z":20,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":7,"y":30,"z":20,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":7,"y":30,"z":23,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":7,"y":30,"z":26,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":0,"y":33,"z":20,"width":3,"length":1,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":0,"y":33,"z":23,"width":3,"length":1,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":0,"y":25,"z":24,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":2,"y":30,"z":23,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":1,"y":30,"z":26,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":1,"y":25,"z":27,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":7,"y":25,"z":27,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":2,"y":25,"z":24,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":2,"y":25,"z":27,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":3,"y":25,"z":24,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":3,"y":25,"z":27,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":4,"y":25,"z":24,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":4,"y":25,"z":27,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":5,"y":25,"z":24,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":5,"y":25,"z":27,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":6,"y":25,"z":24,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":6,"y":25,"z":27,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":7,"y":25,"z":24,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":0,"y":28,"z":24,"width":3,"length":1,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":1,"y":30,"z":23,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":0,"y":28,"z":27,"width":3,"length":1,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":3,"y":28,"z":24,"width":3,"length":1,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":3,"y":28,"z":27,"width":3,"length":1,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":0,"y":29,"z":24,"width":3,"length":1,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":0,"y":29,"z":27,"width":3,"length":1,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":3,"y":29,"z":24,"width":3,"length":1,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":3,"y":29,"z":27,"width":3,"length":1,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":0,"y":25,"z":27,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":0,"y":30,"z":23,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":0,"y":30,"z":26,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":1,"y":30,"z":20,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}},{"x":0,"y":30,"z":20,"width":1,"length":3,"depth":3,"item":{"description":"Widget","width":3,"length":3,"depth":1,"keepFlat":false}}]},{"box":{"reference":"Stock 4 Single Wall","innerWidth":23,"innerLength":30,"innerDepth":30},"items":[{"x":0,"y":0,"z":0,"width":5,"length":30,"depth":5,"item":{"description":"Whatsanamie","width":5,"length":5,"depth":30,"keepFlat":false}}]}]`

**Update**: I forgot to mention: I'm using the InfalliblePacker with `$packer->setMaxBoxesToBalanceWeight(0);`

What's causing this behavior? I.e. what am I overlooking?

// FvW

**Comment by dvdoug** [comment]
I'll look into this one, I would expect that last item to indeed fit

**Reply by frankvw2017** [reply]
Tnx! Let me know if you need me to supply any additional details.

**Reply by dvdoug** [reply, marked as answer]
Fixed in v3.12

**Reply by frankvw2017** [reply]
THANK YOU!!!!
